src/damavand/cloud/aws/resources/glue_component.py

```diff
@@ -118,28 +375,77 @@ def iceberg_database(self) -> aws.glue.CatalogDatabase:
             location_uri=f"s3://{self.iceberg_bucket.bucket}/",
         )
 
+    def _create_glue_trigger(
+        self, job: GlueJobDefinition
+    ) -> Optional[aws.glue.Trigger]:
+        """Creates a Glue Trigger for the job if scheduled.
+
+        Args:
+            job (GlueJobDefinition): The job definition.
+
+        Returns:
+            Optional[aws.glue.Trigger]: The Glue trigger if scheduled, else None.
+        """
+        return (
+            aws.glue.Trigger(
+                f"{job.name}-glue-trigger",
+                type="SCHEDULED",
+                schedule=job.schedule,
+                actions=[
+                    aws.glue.TriggerActionArgs(
+                        job_name=self._get_job_name(job),
+                    )
+                ],
+                start_on_creation=True,
+            )
+            if job.schedule
+            else None
+        )
+
```

Comment:
Optional types are potentials for bugs. It is better to stay away from them as much as possible. I suggest:

```suggestion
    def _create_glue_trigger(
        self, 
        schedule: JobSchedule,
        job_name: str,
    ) -> Optional[aws.glue.Trigger]:
        """Creates a Glue Trigger for the job if scheduled.

        Args:
            job (GlueJobDefinition): The job definition.

        Returns:
            Optional[aws.glue.Trigger]: The Glue trigger if scheduled, else None.
        """
        return  aws.glue.Trigger(
                f"{job.name}-glue-trigger",
                type="SCHEDULED",
                schedule=schedule,
                actions=[
                    aws.glue.TriggerActionArgs(
                        job_name=job_name,
                    )
                ],
                start_on_creation=True,
        )

```

Reply:
Done